hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/tensorflow/config/TensorBoardProcessMgr.java

```diff
@@ -0,0 +1,331 @@
+/*
+ * This file is part of Hopsworks
+ * Copyright (C) 2018, Logical Clocks AB. All rights reserved
+ *
+ * Hopsworks is free software: you can redistribute it and/or modify it under the terms of
+ * the GNU Affero General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * Hopsworks is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.hops.hopsworks.common.dao.tensorflow.config;
+
+import io.hops.hopsworks.common.dao.hdfs.HdfsLeDescriptorsFacade;
+import io.hops.hopsworks.common.dao.hdfsUser.HdfsUsers;
+import io.hops.hopsworks.common.dao.hdfsUser.HdfsUsersFacade;
+import io.hops.hopsworks.common.dao.project.Project;
+import io.hops.hopsworks.common.dao.project.ProjectFacade;
+import io.hops.hopsworks.common.dao.tensorflow.TensorBoard;
+import io.hops.hopsworks.common.dao.tensorflow.TensorBoardFacade;
+import io.hops.hopsworks.common.dao.user.UserFacade;
+import io.hops.hopsworks.common.dao.user.Users;
+import io.hops.hopsworks.common.hdfs.HdfsUsersController;
+import io.hops.hopsworks.common.metadata.exception.DatabaseException;
+import io.hops.hopsworks.common.util.Settings;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.io.FileUtils;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.ejb.ConcurrencyManagement;
+import javax.ejb.ConcurrencyManagementType;
+import javax.ejb.DependsOn;
+import javax.ejb.Singleton;
+import javax.ejb.EJB;
+import javax.ejb.TransactionAttribute;
+import javax.ejb.TransactionAttributeType;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+/**
+ * *
+ * This class wraps a bash script with sudo rights that can be executed by the node['hopsworks']['user'].
+ * /srv/hops/domains/domain1/bin/tfserving.sh
+ * The bash script has several commands with parameters that can be executed.
+ * This class provides a Java interface for executing the commands.
+ */
+@Singleton
```

Comment:
This class is a singleton. If the script returns successfully you have a thread sleep of 5 seconds. This means that if we are 5 different users working on different projects and I'm the unlucky one it will take at least 25 seconds before my request receives a response. 

Beside the terrible user experience, you might have problems as some clients might give up on waiting for the request to be satisfied. Especially if the users are more than 5.

My understanding form the rest of the code is that each project user can have at most 1 TensorBoard open, how can users in different projects or same user in different project can interfere with each other?